src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -13484,7 +13484,8 @@ bool Unit::UpdatePosition(float x, float y, float z, float orientation, bool tel
         return false;
     }
 
-    bool const turn = (GetOrientation() != orientation);
+    // Check if angular distance changed
+    bool const turn = M_PI - fabs(fabs(GetOrientation() - orientation) - M_PI) > 0.001f;
```

Comment:
The code below was changed to deal with cancelling hover spells - 45f8135ac45799e555449bebb6d2a78869a439b6